R/add-subvariable.R

```diff
@@ -12,25 +13,55 @@
 #' }
 #' @export
 addSubvariable <- function (variable, subvariable) {
-    ## Get subvariable URL or URLs, depending on how many supplied
-    new.urls <- urls(subvariable)
+    subvariable <- add_subvar_def(variable, subvariable)
 
-    ## Store these for post workaround
-    subvar.urls <- subvariables(tuple(variable))
+    ## There is some inconsistency in the output order after patching the subvariable catalog
+    ## this stores the proper order.
+    order <- c(names(subvariables(variable)), vapply(subvariable, name, ""))
 
+    ## Get subvariable URL or URLs, depending on how many supplied
+    new.urls <- urls(subvariable)
     ## Do the adding
     crPATCH(shojiURL(variable, "catalogs", "subvariables"),
         body=toJSON(sapply(new.urls, emptyObject, simplify=FALSE)))
-
-    ## Workaround because apparently bind/rebind isn't retaining the order
-    crPATCH(self(variable),
-        body=toJSON(list(subvariables=I(c(subvar.urls, new.urls)))))
-
+    variable <- refresh(variable)
+    subvariables(variable) <- subvariables(variable)[order]
     ## Refresh and return
     dropCache(datasetReference(variable))
-    invisible(refresh(variable))
+    return(invisible(refresh(variable)))
 }
 
 #' @rdname addSubvariable
 #' @export
 addSubvariables <- addSubvariable
+
+add_subvar_def <- function(var, subvar){
```

Comment:
Please camelCase the function name and insert spaces around the () in the function definition, for consistency.